src/hooks/useTable.js

```diff
@@ -230,9 +229,6 @@ export const useTable = (props, ...plugins) => {
   instanceRef.current.rows = rows
   instanceRef.current.flatRows = flatRows
 
-  // Determine column visibility
-  determineHeaderVisibility(instanceRef.current)
-
```

Comment:
It almost seems like we need to add a new hook here, so we can maintain the original order of ops here. I know it's important, still trying to remember why.

Reply:
Yeah...`a new hook` will help in keeping things clear.
`determineHeaderVisibility` enables us to calculate - `Column Visibility` and `visible child columns` which is used in forming `group headers`, `width` calculation in `useAbsoluteLayout`/`useBlockLayout`, this is also re-calculated when columns are re-ordered....